exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_client.h

```diff
@@ -25,12 +29,20 @@ namespace otlp
 
 struct OtlpGrpcClientOptions;
 
+#ifdef ENABLE_ASYNC_EXPORT
+struct OtlpGrpcClientAsyncData;
+#endif
+
 /**
  * The OTLP gRPC client contains utility functions of gRPC.
  */
 class OtlpGrpcClient
 {
 public:
+  OtlpGrpcClient();
```

Comment:
In this class, existing method `OtlpGrpcClient::MakeCompletionQueue()` seems to be dead code.

And beside, it was not protected by ifdef ENABLE_ASYNC_EXPORT.

Please remove.

Reply:
Thanks, and done.

Reply:
My comment about "not protected by ifdef ENABLE_ASYNC_EXPORT" was about method OtlpGrpcClient::MakeCompletionQueue() which is now removed, great.

I see that the constructor and destructor for `OtlpGrpcClient` are now declared with an ifdef, instead of always declaring them and using the ifdef inside the implementation.

To clarify, I think the previous code was better:

in header file:

```
class OtlpGrpcClient
{
public:
  OtlpGrpcClient();
```

in cc file:

```
OtlpGrpcClient::OtlpGrpcClient() {
ifdef ENABLE_ASYNC_EXPORT
... code ...
endif
}
```

This confusion was probably caused by me, commenting on the constructor and saying "not protected by ifdef ENABLE_ASYNC_EXPORT" when talking about MakeCompletionQueue(), sorry about this.

Please use the form you prefer, no need to go back and forth on this, this is minor anyway.

Reply:
Personally, I also prefer the previous one, it produces the same symbols with or without async mode.

`OtlpGrpcClient::MakeCompletionQueue()` is used to ensure the symbols in headers of gRPC are produced in otlp_grpc_client. I'm not sure but they may also be created in `DelegateExport()`.

I will change the constructor and destructor to previous implementation.

Reply:
The last commit reverted the fix for the wait_for condition in the loop.

Should be running_requests == 0.

Reply:
Sorry, it's fixed now.